clickhouse_search/managers.py

```diff
@@ -755,39 +768,14 @@ def _search_call_data(self, entries, sample_data, inheritance_mode=None, inherit
        inheritance_q = None
        quality_q = None
        gt_filter_map = None
+       family_missing_type_samples = None
        quality_filter = qualityFilter or {}
        individual_genotype_filter = (inheritance_filter or {}).get('genotype')
-       custom_affected = (inheritance_filter or {}).get('affected') or {}
        if inheritance_mode or individual_genotype_filter or quality_filter:
-            clinvar_override_q = AnnotationsQuerySet._clinvar_path_q(
-               pathogenicity, _get_range_q=lambda path_range: Q(clinvar_join__pathogenicity__range=path_range),
-            ) if self._has_clinvar() else None
-
-            family_sample_gts = {}
-            family_sample_null_gts = {}
-            family_affected_samples = {}
-            family_quality_samples = {}
-            family_missing_type_samples = {}
-            for s in sample_data:
-                sample_filters, any_affected_samples, nullable_gt_samples, quality_samples, missing_type_samples = self._family_sample_filters(
-                    s, inheritance_mode, individual_genotype_filter, quality_filter, custom_affected,
-                    is_multi_sample_family=s['family_guid'] in multi_sample_type_families,
-                )
-                if sample_filters:
-                    family_sample_gts[s['family_guid']] = sample_filters
-                if any_affected_samples:
-                    family_affected_samples[s['family_guid']] = any_affected_samples
-                if nullable_gt_samples:
-                    family_sample_null_gts[s['family_guid']] = nullable_gt_samples
-                if quality_samples:
-                    family_quality_samples[s['family_guid']] = quality_samples
-                if missing_type_samples:
-                    family_missing_type_samples[s['family_guid']] = missing_type_samples
-
-            is_single_family = len(sample_data) == 1 and not multi_sample_type_families
-            inheritance_q, gt_filter_map = self._inheritance_q(family_sample_gts, family_sample_null_gts, family_affected_samples, is_single_family)
-            quality_q = self._quality_q(quality_filter, family_quality_samples, clinvar_override_q, is_single_family)
-
+            inheritance_q, quality_q, gt_filter_map, family_missing_type_samples = self._get_inheritance_quality_qs(
```

Comment:
abstracted this code block into a helper function with no ther changes to resolve codacy flagging this function as too complex